src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -153,26 +175,56 @@ ucc_tl_ucp_reduce_scatter_knomial_progress(ucc_coll_task_t *coll_task)
                      (avg_pre_op ? ucc_knomial_pattern_loop_first_iteration(p)
                                  : ucc_knomial_pattern_loop_last_iteration(p));
 
-            status = ucc_tl_ucp_reduce_multi(
+            status = ucc_tl_ucp_reduce_multi_nb(
                 local_data, rbuf, reduce_data,
                 task->tagged.send_posted - p->iteration * (radix - 1),
-                local_seg_count, local_seg_count * dt_size, dt, mem_type, task,
-                is_avg);
+                local_seg_count, local_seg_count * dt_size, dt, task,
+                is_avg, exec, &task->reduce_scatter_kn.etask);
             if (ucc_unlikely(UCC_OK != status)) {
                 tl_error(UCC_TASK_LIB(task), "failed to perform dt reduction");
                 task->super.super.status = status;
                 return status;
             }
+UCC_KN_PHASE_REDUCE:
+            if (task->reduce_scatter_kn.etask != NULL) {
```

Comment:
Is it possible to make this code (everything in that "if" block) macro or inline fn? it is used in multiple locations

Reply:
done